### PR TITLE
feat: Search categories

### PR DESCRIPTION
**Pull request by krgarima**
## Change Logs:
# Description: Added Search feature

# More details:

1. Search -
- User can search for any category by entering categories in the search bar. Categories will be displayed on the same Home page below.
quizSearch

2. Some code refactoring.

Screenshot:

### CODE REVIEW

File: src/Components/Navbar/Navbar.js [hunk 1]
```diff
@@ -1,18 +1,48 @@
...
   return (
-    <div className="navbar space-between">
-      <span className="left space-between">
-        <h3 className="third-heading company-logo">
-          <Link to="/">Quizwiz</Link>
-        </h3>
-        <i className="fas fa-search"></i>
-      </span>
+    <div className="navbar">
+      {console.log(successMsg)}
```
Comment (Asuraking1n) [inline src/Components/Navbar/Navbar.js]: Remove useless codes

**Review by Asuraking1n — APPROVED** [review]
Looks good...

File: src/Components/Card/Card.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
 import React, { useContext } from "react";
-import { CategoryContext } from "../../Context/category-context";
+import { CategoryContext, SearchContext } from "../../Context/index";
 import { Link } from "react-router-dom";
 import { categories } from "../../backend/db/categories";
```
Comment (arpit01923) [inline src/Components/Card/Card.js:4]: use absolute path to avoid ../../

File: src/Components/Navbar/Navbar.js [hunk 3]
```diff
@@ -1,18 +1,47 @@
-import React, { useContext } from "react";
-import "./Navbar.css";
+import React, { useContext, useState } from "react";
 import { Link } from "react-router-dom";
-import { AuthContext } from "../../Context/auth-context";
+import { AuthContext, SearchContext } from "../../Context/index";
```
Comment (arpit01923) [inline src/Components/Navbar/Navbar.js:3]: no need of writing /index

**Review by Balakumar3018 — APPROVED** [review]
UI looks clean and functionality is working as expected. Good to merge

File: src/Pages/Home/Home.css [hunk 4]
```diff
@@ -40,12 +40,13 @@
   -webkit-text-stroke: 2px var(--primary-color);
```
Comment (arpit01923) [inline src/Pages/Home/Home.css:40]: want to know how we used (-)before variable in css

File: src/Pages/Home/Home.js [hunk 5]
```diff
@@ -1,21 +1,35 @@
-import React from "react";
+import React, { useContext } from "react";
 import Card from "../../Components/Card/Card";
```
Comment (arpit01923) [inline src/Pages/Home/Home.js:2]: use consistent imports, either used naming or default imports

File: src/Pages/Result/Result.js [hunk 6]
```diff
@@ -1,8 +1,6 @@
 import React, { useContext, useState, useEffect } from "react";
 import { Link } from "react-router-dom";
-import { ScoreContext } from "../../Context/totalScore-context";
-import { NameContext } from "../../Context/name-context";
-import { ResultContext } from "../../Context/result-context";
+import { ScoreContext, NameContext, ResultContext } from "../../Context/index";
```
Comment (arpit01923) [inline src/Pages/Result/Result.js:3]: same use absolute path to avoid ../../

**Review by arpit01923 — COMMENTED** [review]
- overall code is good
- naming convention are good
- left few suggestions

**Pull request merged**